lib/rucio/common/utils.py

```diff
@@ -611,18 +611,18 @@ def default(self, obj):  # pylint: disable=E0202
         return json.JSONEncoder.default(self, obj)
 
 
-def render_json(**data):
-    """ JSON render function
+def render_json(list_: Optional[list[Any]] = None, **dict_) -> str:
```

Comment:
By using `list_` as a named argument, there is still the risk of a `TypeError`, as mentioned in the beginning here: https://github.com/rucio/rucio/pull/6648#discussion_r1596781366

Please rewrite the method like this:
```python
def render_json(*args, **kwargs) -> str:
    """ Render a list or a dict as a JSON-formatted string. """
    if args and isinstance(args[0], list):
        data = args[0]
    elif kwargs:
        data = kwargs
    else:
        raise ValueError("Error while serializing object to JSON-formatted string: supported input types are list or dict.")
    return json.dumps(data, cls=APIEncoder)
```